### ISSUE TITLE
BT-39: Only build pull requests

### ISSUE THREAD

**Issue by ataylorme** [pantheon-systems/terminus-build-tools-plugin#118]
When setting up CircleCI I think we should set the only build pull request option when scaffolding a new project to fix our failing Behat jobs when a mulidev doesn't deploy due to no pull request being open

![circle-ci-only-build-pull-requests](https://user-images.githubusercontent.com/2133004/32455864-755139ee-c2d8-11e7-9a49-05144adc195c.png)

**Comment by greg-1-anderson** [comment]
I don't see a way to set that option in the [Circle CI Project methods](https://circleci.com/docs/api/v1-reference/#projects). Is there perhaps some other section of the Circle API that exposes this bit?

I have the example-drops-8-composer project configured to make a `ci-*` build if no pull request has been created yet. This, at least, avoids a failed build. I'm not opposed to setting that option, if there's a way to do it.

**Comment by stevector** [comment]
This worked for me:

```
curl -H 'Content-Type: application/json' -X PUT \
-d '{"feature_flags": {"build-prs-only": true}}' \
https://circleci.com/api/v1.1/project/github/pantheon-systems/pantheonio/settings\?circle-token\=XXXXXXXXXX
```

**Comment by ElijahLynn** [comment]
Confirmed this works for me too!

From what I gather, a new method needs to be added to https://github.com/pantheon-systems/terminus-build-tools-plugin/blob/master/src/ServiceProviders/CIProviders/CircleCI/CircleCIProvider.php for the feature_flags PUT. Maybe a `CircleCIProvider->setFeatureFlags()` and then refactor `circleCIAPI()` to also do a PUT for the feature flags. I did try it as a POST and it just returned HTML. 

@stevector Where did you find this API information or how did you discover it? The API docs don't mention that endpoint, 'feature_flags' or even any PUT methods.

**Comment by stevector** [comment]
@ElijahLynn, that would probably work!

I found that URL just by watching Chrome Dev Tools while using that UI. It might not be a part of the official API but I don't think that needs to stop us from using it.

**Comment by ElijahLynn** [comment]
Nice sleuthing! 

I would just make sure to add a comment around that in the new method and that it is undocumented and could break at any time.

**Comment by greg-1-anderson** [comment]
Fixed by #191 (at least for CircleCI)
